### PR TITLE
use `map` instead of `forEach`

### PR DESCRIPTION
**Pull request by anuragasaurus**
use `map` instead of `forEach` if not returning a new array.

### CODE REVIEW

**Comment by reznord** [comment]
Why are you doing a `forEach` instead of a `map`? 

_Sent from my OnePlus ONEPLUS A3003 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by siddharthkp** [comment]
@anuragasaurus i understand that it's not the primary case map is used for, but is there any advantage of using forEach over map here?

**Comment by thiamsantos** [comment]
I think that the correct approach would be use `reduce`. Something like:

```js
// ...
.then(response => response.data.reduce((values, file) => {
  values[file.path] = file.size
  return values
}, {}))
// ...
```

**Comment by hzlmn** [comment]
@siddharthkp I am not a `v8` engineer, but as far as i know from some of them - `forEach` internally has a bit better performance. But it has not significant differences on small data. But in this case IMO, readability should come in first place.

**Comment by siddharthkp** [comment]
@anuragasaurus @thiamsantos Can you point me to docs/article pointed out the use cases for map vs forEach vs reduce? I don't know enough to take a call.

**Comment by thiamsantos** [comment]
## [Foreach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)
Foreach takes a callback function and run that callback function on each element of array one by one. Basically forEach works as a traditional for loop looping over the array and providing you array elements to do operations on them.

## [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)
Map generates a new array based on your existing array by transforming the values of the array.

## [Reduce](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)
Reduce reduces the array to one single value.

## tl;dr
- forEach doesn’t return **anything**
- map returns a **new array**
- reduce returns a **single value**, that could be a number, an object or any other value including an array

### Fonts:
- https://medium.com/@manojsinghnegi/array-methods-explained-filter-vs-map-vs-reduce-vs-foreach-ea3127c6d319
- https://www.discovermeteor.com/blog/understanding-javascript-map/
- http://cryto.net/~joepie91/blog/2015/05/04/functional-programming-in-javascript-map-filter-reduce/
- https://medium.com/javascript-scene/reduce-composing-software-fe22f0c39a1d

**Comment by siddharthkp** [comment]
@thiamsantos Thanks so much! 

It seems like the return type does not make a difference on the implementation because I set the size in a different object altogether `values`

However, I see that I'm mutating `response.data` for no reason. Which is bad, but not a big problem because it is contained in the function (side effects don't leak out)

From a being accurate standpoint, `forEach` seems perfect, from a readability standpoint, i like `map` more. Because, the side effects creating do not cause a bug/problem in the function (I don't see future scope as well), I am leaning towards readability instead of correctness. But that's just me.

What are your thoughts @anuragasaurus @thiamsantos @hzlmn @reznord

**Comment by anuragasaurus** [comment]
@siddharthkp it was just a small suggestion. You should choose whatever is more readable to you.

**Comment by siddharthkp** [comment]
@anuragasaurus sorry for going all analysis paralysis on your PR, just trying to learn 🙂

**Comment by anuragasaurus** [comment]
`analysis paralysis` :) . hey, no problem.
